RHIDP-2908 + RHIDP-2895: Remove k8s from OCP section, split k8s by platform (#328)
redhat-developer/red-hat-developers-documentation-rhdh: modules/admin/proc-rhdh-deploy-eks-using-helm.adoc

// Module included in the following assemblies:
//
// * assemblies/assembly-rhdh-integration-aws.adoc
//

[id="proc-rhdh-deploy-eks-using-helm_{context}"]
= Deploying {product} in {eks-name} ({eks-short}) using Helm Chart

When you deploy {product-short} in {eks-name} ({eks-short}) using Helm Chart, it orchestrates a robust development environment within the AWS ecosystem.

.Prerequisites

* You have an {eks-short} cluster with AWS Application Load Balancer (ALB) add-on installed. For more information, see https://docs.aws.amazon.com/eks/latest/userguide/alb-ingress.html[Application load balancing on Amazon {product-short}] and https://docs.aws.amazon.com/eks/latest/userguide/aws-load-balancer-controller.html[Installing the AWS Load Balancer Controller add-on].
* You have installed the link:https://kubernetes.io/docs/reference/kubectl/[`kubectl` CLI].

> * You started a `kubectl` session, with `developer` or `admin` permissions.

Reviewer: do you think this would be more clear as...
```suggestion
* You are logged into your cluster using `kubectl`, and have `developer` or `admin` permissions.
```

* You have an entry in the AWS Certificate Manager (ACM) for your preferred domain name. Make sure to keep a record of your Certificate ARN.
* You have subscribed to `registry.redhat.io`. For more information, see https://access.redhat.com/RegistryAuthentication[{company-name} Container Registry Authentication].
* You have set the context to the {eks-short} cluster in your current `kubeconfig`. For more information, see https://docs.aws.amazon.com/eks/latest/userguide/create-kubeconfig.html[Creating or updating a kubeconfig file for an Amazon {eks-short} cluster].
* You have installed Helm 3 or the latest. For more information, see https://docs.aws.amazon.com/eks/latest/userguide/helm.html[Using Helm with Amazon {eks-short}].

.Procedure

. Go to your terminal and run the following command to add the Helm chart repository containing the {product-short} chart to your local Helm registry:
+
[source,terminal]
----
helm repo add openshift-helm-charts https://charts.openshift.io/
----

. Create and activate the _<rhdh>_ namespace:
+
[source,terminal,subs="quotes+"]

> ----
> NAMESPACE=_<rhdh>_

Author: ```suggestion
----
DEPLOYMENT_NAME=_redhat-developer-hub_
NAMESPACE=_<rhdh>_
```

@nickboldt right?

> NAMESPACE=_<rhdh>_

Reviewer: ```suggestion
DEPLOYMENT_NAME=_<redhat-developer-hub>_
NAMESPACE=_<rhdh>_
```

  volumePermissions:
    enabled: true
----

. Run the following command in your terminal to deploy {product-short} using the latest version of Helm Chart and using the `values.yaml` file created in the previous step:
+
[source,terminal,subs="attributes+"]
----
helm install rhdh \
  openshift-helm-charts/redhat-developer-hub \
  [--version {product-chart-version}] \
  --values /path/to/values.yaml
----
+
[NOTE]
====
For the latest chart version, see https://github.com/openshift-helm-charts/charts/tree/main/charts/redhat/redhat/redhat-developer-hub
====

> . Configure your {product-short} Helm chart instance with the {product-short} database password and router base URL values from your {ocp-short} cluster:

Reviewer: ```suggestion
. Configure your {product-short} Helm chart instance with the {product-short} database password and router base URL values from your cluster:
```

+
[source,terminal,subs="attributes+"]
----
PASSWORD=$(kubectl get secret redhat-developer-hub-postgresql -o jsonpath="{.data.password}" | base64 -d)
CLUSTER_ROUTER_BASE=$(kubectl get route console -n openshift-console -o=jsonpath='{.spec.host}' | sed 's/^[^.]*\.//')

> helm upgrade redhat-developer-hub -i "https://github.com/openshift-helm-charts/charts/releases/download/redhat-redhat-developer-hub-{product-chart-version}/redhat-developer-hub-{product-chart-version}.tgz" \

Author: ```suggestion
helm upgrade $DEPLOYMENT_NAME -i "https://github.com/openshift-helm-charts/charts/releases/download/redhat-redhat-developer-hub-{product-chart-version}/redhat-developer-hub-{product-chart-version}.tgz" \
```
@nickboldt correct?

    --set global.clusterRouterBase="$CLUSTER_ROUTER_BASE" \
    --set global.postgresql.auth.password="$PASSWORD"
----

. Display the running {product-short} instance URL, by running the following command:
+
[source,terminal]
----

> echo "https://redhat-developer-hub-$NAMESPACE.$CLUSTER_ROUTER_BASE"

Author: ```suggestion
echo "https://$DEPLOYMENT_NAME-$NAMESPACE.$CLUSTER_ROUTER_BASE"
```
@nickboldt correct?

> echo "https://redhat-developer-hub-$NAMESPACE.$CLUSTER_ROUTER_BASE"

Reviewer: ```suggestion
echo "https://$DEPLOYMENT_NAME-$NAMESPACE.$CLUSTER_ROUTER_BASE"
```
